services/repository/create.go

```diff
@@ -240,13 +241,13 @@ func CreateRepositoryDirectly(ctx context.Context, doer, u *user_model.User, opt
 		ObjectFormatName:                opts.ObjectFormatName,
 	}
 
-	var rollbackRepo *repo_model.Repository
+	needsUpdateStatus := opts.Status == 0
```

Comment:
```suggestion
	needsUpdateStatus := opts.Status == repo_model.RepositoryReady
```

Reply:
d66b9059a2648bf224af5c669692a864344abbf9

And the previous logic is wrong. It should be `opts.Status != repo_model.RepositoryReady`.

Reply:
I also confused here.